src/x360.rs

```diff
@@ -316,6 +332,59 @@ impl<CL: Borrow<Client>> Xbox360Wired<CL> {
 			}
 		}
 	}
+
+	/// Register notification handler.
+	#[inline(never)]
+	pub fn register_notification(&mut self, handler: Box<dyn Xbox360WiredNotificationHandler>) -> Result<(), Error> {
+		{
+			let mut notification_handler = self.notification_handler.lock().unwrap();
+			*notification_handler = Some(handler);
+		}
+		let thread_notification_handler = self.notification_handler.clone();
+
+		let serial_no = self.serial_no;
```

Comment:
This kind of corner case must be handled you're right. I'm quite new in the Rust world, and still with the C mindset "The langage won't help you, you always have to be careful".

I'm not sure this case is possible, because at every plug you increment the `serialno`. So you can't register for the notifications of the same controller twice. 

About the case where the same `Xbox360Wired` is kept with the following sequence
1. `register_notification()`
2. `unregister_notification()`
3. `register_notification()`

There should be no issue if the thread is correctly joined. The current implementation has no correct error handling, it should be improved.